agent/tcs/client/client.go

```diff
@@ -242,6 +263,44 @@ func (cs *clientServer) metricsToPublishMetricRequests() ([]*ecstcs.PublishMetri
 	return requests, nil
 }
 
+// serviceConnectMetricsToPublishMetricRequests loops over all the SC metrics in a
+// task metric to add SC metrics until the message size is within 1 MB.
+// If adding a SC metric to the message exceeds the 1 MB limit, it will be sent in the new message
+func (cs *clientServer) serviceConnectMetricsToPublishMetricRequests(requestMetadata *ecstcs.MetricsMetadata, taskMetric *ecstcs.TaskMetric,
+	messageTaskMetrics []*ecstcs.TaskMetric, requests []*ecstcs.PublishMetricsRequest) (*ecstcs.TaskMetric, []*ecstcs.TaskMetric, []*ecstcs.PublishMetricsRequest) {
+	tempTaskMetric := *taskMetric
+	tempTaskMetric.ServiceConnectMetricsWrapper = tempTaskMetric.ServiceConnectMetricsWrapper[:0]
+
+	for j, serviceConnectMetric := range taskMetric.ServiceConnectMetricsWrapper {
```

Comment:
I don't think this loop is doing anything? you're emptying out `tempTaskMetric.ServiceConnectMetricsWrapper` in linve 277 above.

Reply:
hmm I don't understand....I think we need this loop to incrementally add the service connect metrics to temp  variable to keep the request size in check(probably, I misunderstood your question here)

Reply:
Oh, nvm. Just realized the loop is over `taskMetric.ServiceConnectMetricsWrapper`. I was reading it as if it was over `tempTaskMetric.ServiceConnectMetricsWrapper`